server/src/main/java/de/uftos/services/StudentService.java

```diff
@@ -39,7 +40,28 @@ public StudentService(StudentRepository repository) {
    */
   public Page<Student> get(Pageable pageable, Optional<String> firstName,
                            Optional<String> lastName, Optional<String[]> tags) {
-    return this.repository.findAll(pageable);
+    Specification<Student> spec = createSpecification(firstName, lastName, tags);
+    return this.repository.findAll(spec, pageable);
+  }
+
+  private Specification<Student> createSpecification(Optional<String> firstName,
+                                                         Optional<String> lastName,
+                                                         Optional<String[]> tags) {
+    Specification<Student> spec = Specification.where(null);
+
+    firstName.ifPresent(param -> spec.or(createFilter(param, "firstName")));
+    lastName.ifPresent(param -> spec.or(createFilter(param, "lastName")));
+    tags.ifPresent(param -> {
+      for (String tag : param) {
+        spec.or(createFilter(tag, "tags"));
```

Comment:
tags filter should be connected with and, because tags are filtered with a dropdown menu and only students should be shown which have all selected tags.
CreateFilter method not usable for tags, as ids of tags are given to filter.
A hopefully working version of a filter for tags should soon be available in PR #97.

Reply:
I tested your proposition and it didn't work, but mine didn't either. Daniel and I found the correct way of doing it